### PR TITLE
New kernel for XL-BOMD and fix bug in response_orig

### PR DESCRIPTION
**Pull request by ares201005**
Add a new Kernel function for XL-BOMD in LATTE and fix a bug in prg_canon_response_orig.

---
This change is [Reviewable](https://reviewable.io/reviews/lanl/qmd-progress/216)

### CODE REVIEW

File: src/prg_response_mod.F90 [hunk 1]
```diff
@@ -831,6 +832,107 @@ subroutine prg_project_response(rsp_bml,over_bml,spindex,norbi,coordinates,rspfu
...
+    real(dp)              :: h_0(HDIM), p_0(HDIM), dPdmu(HDIM), p_02(HDIM),iD0(HDIM)
+    real(dp)              :: cnst, kB, mu1
+    real(dp), allocatable :: FOCC(:), row1(:)
+    real(dp)              :: tmp, tmp2, dmu
+    integer               :: i, j, k
+    character(20) :: bml_type
+
+    kB = 8.61739e-5        ! (eV/K)
+
+    ALLOCATE(FOCC(HDIM))
```
Comment (nicolasbock) [inline src/prg_response_mod.F90:868]: Please convert Fortran keywords to lower case, i.e.

```suggestion
    allocate(FOCC(HDIM))
```

**Review by nicolasbock — CHANGES_REQUESTED** [review]

File: src/prg_response_mod.F90 (hunk 1 above)
Comment (ares201005) [reply src/prg_response_mod.F90:868]: Ok, thanks. Just found there is new merge, need to rebase as well.

**Review by cnegre — APPROVED** [review]

**Review by nicolasbock — APPROVED** [review]
Thanks!

File: src/prg_response_mod.F90 [hunk 2]
```diff
@@ -831,6 +832,105 @@ subroutine prg_project_response(rsp_bml,over_bml,spindex,norbi,coordinates,rspfu
...
+      tmp = exp((evals(i) - mu0) * beta)
+      tmp2 = (1.D0 + tmp) * (1.D0 + tmp)
+      dPdmu(I) = -tmp/tmp2 * beta
+    enddo
+
+    do I = 1, HDIM
+      X(I,I) = X(I,I) + 1.D0
+    enddo
+
+    do I = 1, HDIM
```
[the remark this answers is not in the document]
Comment (ares201005) [reply src/prg_response_mod.F90:890]: Right, will do it.

[the remark this answers is not in the document]
Comment (nicolasbock) [reply src/prg_response_mod.F90:890]: @jeanlucf22 Are you ok with merging this PR as is or would you prefer the work on the BML to be added first?

[the remark this answers is not in the document]
Comment (jeanlucf22) [reply src/prg_response_mod.F90:890]: @nicolasbock go ahead! We want that function integrated ASAP. I was just commenting about what I see as a future bottleneck

[the remark this answers is not in the document]
Comment (nicolasbock) [reply src/prg_response_mod.F90:890]: :+1:

**Comment by nicolasbock** [comment]
I created lanl/bml/#592 to track the refactor @ares201005

**Comment by ares201005** [comment]
Thank you!

**Pull request merged**